src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs

```diff
@@ -67,6 +71,26 @@ public static void AddTagsFromDbCommand(this Span span, IDbCommand command)
             }
         }
 
+        internal static void SetHeaderTags(this Span span, IHeadersCollection headers, IDictionary<string, string> headerTags, string defaultTagPrefix)
+        {
+            if (headerTags is not null && !headerTags.IsEmpty())
+            {
+                try
+                {
+                    // extract propagation details from http headers
```

Comment:
Looks like this comment was copied from somewhere that used `SpanContextPropagator` for something else.

Which made realize that `SpanContextPropagator` is not the right place to copy http header to span tags (we're not propagating any span context), but that's a separate issue 😅

Reply:
There's a good OTel PR to improve context propagation which removes the header tags from the propagation helpers. Hopefully I can port that over soon...Anyways this comment has been removed in c1ac5ce1868b7c359e741bd04d05e742647a9a33